request_example_test.go

```diff
@@ -302,6 +309,18 @@ func (f *memFile) WriteAt(p []byte, off int64) (int, error) {
 	return len(p), nil
 }
 
+func (f *memFile) Truncate(size int64) error {
+	f.contentLock.Lock()
+	defer f.contentLock.Unlock()
+	nc := make([]byte, size)
```

Comment:
We should be able to truncate the byte slice if `size` is shorter, and just `append(f.content, make([]byte, grow)...)` otherwise:

```
grow := size - len(f.content)
if grow <= 0 {
  f.content = f.content[:size]
} else {
  f.content = append(f.content, make([]byte, grow)...)
}
```

This uses some [Slice Tricks](https://github.com/golang/go/wiki/SliceTricks#extend) that the compiler understands, and will turn into “just expand, don’t actually allocate the `make` buffer”. [The source code](https://github.com/golang/go/blob/037ac2bd84480e0a06c4d1e7a2c1f133109466a5/src/cmd/compile/internal/gc/walk.go#L662-L664)